src/allmydata/grid_manager.py

```diff
@@ -0,0 +1,348 @@
+import sys
+import json
+from datetime import (
+    datetime,
+    timedelta,
+)
+
+from allmydata.crypto import (
+    ed25519,
+)
+from allmydata.util import (
+    fileutil,
+    base32,
+)
+
+
+class _GridManagerStorageServer(object):
+    """
+    A Grid Manager's notion of a storage server
+    """
+
+    def __init__(self, name, public_key, certificates):
+        self.name = name
+        self._public_key = public_key
+        self._certificates = [] if certificates is None else certificates
+
+    def add_certificate(self, certificate):
+        self._certificates.append(certificate)
+
+    def public_key(self):
+        return ed25519.string_from_verifying_key(self._public_key)
+
+    def marshal(self):
+        return {
+            u"public_key": self.public_key(),
+        }
+
+
+def create_grid_manager():
+    """
+    Create a new Grid Manager with a fresh keypair
+    """
+    private_key, public_key = ed25519.create_signing_keypair()
+    return _GridManager(
+        ed25519.string_from_signing_key(private_key),
+        {},
+    )
+
+
+def load_grid_manager(config_path, config_location):
+    """
+    Load a Grid Manager from existing configuration.
+
+    :param FilePath config_path: the configuration location (or None for
+        stdin)
+
+    :param str config_location: a string describing the config's location
+
+    :returns: a GridManager instance
+    """
+    if config_path is None:
+        config_file = sys.stdin
+    else:
+        try:
+            config_file = config_path.child("config.json").open("r")
+        except IOError:
+            raise ValueError(
+                "'{}' is not a Grid Manager config-directory".format(config_path)
+            )
```

Comment:
Related to the comment above about letting view code deal with presenting errors to users, this could be something like `raise BadConfigDirectory(e)` and the caller can describe the problem.

Also (independent of that) `IOError` is too broad for this particular description.  Maybe it's permission denied or out of file descriptors or interrupted or the path name is too long, etc.